### ISSUE TITLE
Azure gpt-4-turbo-2024-04-09 没有响应，响应被截断或中断了。[Bug] 

### ISSUE THREAD

**Issue by jmartinezb3** [ChatGPTNextWeb/ChatGPT-Next-Web#4624]
### Bug Description

在这个例子中，当请求模型定义图像时，模型没有完全回应，只显示了部分回应。我已经调整了最大生成令牌数参数来检查是否是这个问题，但它仍然不起作用。我还修改了在Azure上的模型部署参数，但我仍然得到相同的结果。这让我认为这可能是用户界面的问题，而不是模型配置的问题。

### Steps to Reproduce

像往常一样使用模型

### Expected Behavior

完整的答案。

### Screenshots

![Test2](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/94153214/28085cf1-acdb-4846-9a24-85957006c862)

![Test](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/94153214/a3a49bca-e239-4940-9b45-7beb622eef8d)

### Deployment Method

- [ ] Docker
- [ ] Vercel
- [ ] Server

### Desktop OS

Windows 10, 桌面应用程序. v2.12.2

**Comment by Dean-YZG** [comment]
my Azure gpt-4-turbo-2024-04-09 doesnt  even support image

**Comment by Dean-YZG** [comment]
currentlly, Azure gpt-4-turbo-2024-04-09 doesnt  support image
https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/gpt-with-vision?tabs=rest%2Csystem-assigned%2Cresource

**Comment by jmartinezb3** [comment]
> my Azure gpt-4-turbo-2024-04-09 doesnt even support image

This is not true. If you call the API with a Python or JS script, it does support images. It is literally one of the main functions of the model.

**Comment by doherty88** [comment]
我也遇到这样的问题，用的是azure平台的gtp-4-turbo，文本问题回答ok的，只要上传图片的回答，答案会被截断

**Comment by jmartinezb3** [comment]
> 我也遇到这样的问题，用的是azure平台的gtp-4-turbo，文本问题回答ok的，只要上传图片的回答，答案会被截断

@doherty88 这似乎是关于如何处理带有图像的请求的问题。当模型需要使用其视觉能力时，图形界面无法正确处理请求。在Azure playground中，它可以被正确使用，所以这不是模型或Azure配置的问题。

**Comment by libli** [comment]
同样问题，我对比了一下，看起来是流式响应时，第二段及后续的响应和openai官方api响应的结构不一样。

**Comment by Dean-YZG** [comment]
> 同样问题，我对比了一下，看起来是流式响应时，第二段及后续的响应和openai官方api响应的结构不一样。

有响应的截图么？

**Comment by Dean-YZG** [comment]
> > my Azure gpt-4-turbo-2024-04-09 doesnt even support image
> 
> This is not true. If you call the API with a Python or JS script, it does support images. It is literally one of the main functions of the model.
> 
> Furthermore, if it didn't support images, in the screenshot I sent, the model wouldn't know what I'm talking about. However, that's not the case. The issue is that the responses are not being displayed correctly. @Dean-YZG

 i agree with you, do you have any screenshots of the response?

**Comment by jmartinezb3** [comment]
Hello, yes, @Dean-YZG 

This is the response using custom proxy from https://github.com/haibbo/cf-openai-azure-proxy

这是使用 https://github.com/haibbo/cf-openai-azure-proxy 提供的自定义代理的响应。

![imagen](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/94153214/6971f6b2-8ec2-4b31-8619-e66ee9cee8ac)
![imagen](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/94153214/14f27512-96ff-424d-a0fe-b999c16311fa)

This is the response using usual Azure API configuration.

这是使用通常的 Azure API 配置的响应。

![imagen](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/94153214/570dfacd-400f-4fc7-9420-a8e736c464dd)
![imagen](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/94153214/2f3b26dc-5670-4a8e-9c2a-74d9913feb10)

**Comment by dustookk** [comment]
I just solved the same problem; I hope this can help you guys.

change the code of [isVisionModel](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/blob/ffe32694b0cb2c832337a074e0ef4b277d7db3ee/app/utils.ts#L251) made sure your model name is included
